examples/api/lib/widgets/application_lifecycle/application_lifecycle.0.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for [ApplicationLifecycleListener].
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  const MyApp({super.key});
+
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: Scaffold(body: Body()),
+    );
+  }
+}
+
+class Body extends StatefulWidget {
+  const Body({super.key});
+
+  @override
+  State<Body> createState() => _BodyState();
+}
+
+class _BodyState extends State<Body> {
+  late final AppLifecycleListener listener;
+  bool _shouldExit = false;
+
+  @override
+  void initState() {
+    super.initState();
+    listener = AppLifecycleListener(
+      binding: WidgetsBinding.instance,
+      onExitRequested: _handleExitRequest,
+    );
+  }
+
+  Future<void> _quit() async {
+    await ServicesBinding.instance.exitApplication(AppExitType.cancelable);
```

Comment:
I think this should be:
```dart
await ServicesBinding.instance.exitApplication(_shouldExit ? AppExitType.cancelable : AppExitType.required);
```
Then you can test a required exit.

Reply:
Or perhaps the other way around? Or maybe a second quit button (quit required).

Reply:
Oh, yes, I guess so.  But we definitely need to hook up the close button to request a cancelable exit in that case, or you won't be able to test that.